src/main/kotlin/cacophony/regex/RegexStrings.kt

```diff
@@ -0,0 +1,88 @@
+        )
+
+    private val specificCategoryMap =
+        mapOf(
+            TokenCategorySpecific.LEFT_PARENTHESIS to """\(""",
+            TokenCategorySpecific.RIGHT_PARENTHESIS to """\)""",
+            TokenCategorySpecific.LEFT_BRACKET to """[""",
+            TokenCategorySpecific.RIGHT_BRACKET to """]""",
+            TokenCategorySpecific.ARROW to """->""",
+            TokenCategorySpecific.DOUBLE_ARROW to """=>""",
+            TokenCategorySpecific.COLON to """:""",
+            TokenCategorySpecific.SEMICOLON to """;""",
+            TokenCategorySpecific.COMMA to """,""",
+            // keywords
+            TokenCategorySpecific.KEYWORD_LET to """let""",
+            TokenCategorySpecific.KEYWORD_IF to """if""",
+            TokenCategorySpecific.KEYWORD_THEN to """then""",
+            TokenCategorySpecific.KEYWORD_ELSE to """else""",
+            TokenCategorySpecific.KEYWORD_WHILE to """while""",
+            TokenCategorySpecific.KEYWORD_DO to """do""",
+            TokenCategorySpecific.KEYWORD_BREAK to """break""",
+            TokenCategorySpecific.KEYWORD_RETURN to """return""",
+            // operators
+            TokenCategorySpecific.OPERATOR_EQUALS to """==""",
+            TokenCategorySpecific.OPERATOR_NOT_EQUALS to """!=""",
+            TokenCategorySpecific.OPERATOR_ASSIGNMENT to """=""",
+            TokenCategorySpecific.OPERATOR_LESS to """<""",
+            TokenCategorySpecific.OPERATOR_GREATER to """>""",
+            TokenCategorySpecific.OPERATOR_LESS_EQUAL to """<=""",
+            TokenCategorySpecific.OPERATOR_GREATER_EQUAL to """>=""",
+            TokenCategorySpecific.OPERATOR_ADDITION to """+""",
+            TokenCategorySpecific.OPERATOR_SUBTRACTION to """-""",
+            TokenCategorySpecific.OPERATOR_ADDITION_ASSIGNMENT to """+=""",
+            TokenCategorySpecific.OPERATOR_SUBTRACTION_ASSIGNMENT to """-=""",
+            TokenCategorySpecific.OPERATOR_MULTIPLICATION to """\*""",
+            TokenCategorySpecific.OPERATOR_DIVISION to """/""",
+            TokenCategorySpecific.OPERATOR_MODULO to """%""",
+            TokenCategorySpecific.OPERATOR_MULTIPLICATION_ASSIGNMENT to """\*=""",
+            TokenCategorySpecific.OPERATOR_DIVISION_ASSIGNMENT to """/=""",
+            TokenCategorySpecific.OPERATOR_MODULO_ASSIGNMENT to """%=""",
+            TokenCategorySpecific.OPERATOR_LOGICAL_OR to """\|\|""",
+            TokenCategorySpecific.OPERATOR_LOGICAL_AND to """&&""",
+            TokenCategorySpecific.OPERATOR_LOGICAL_NOT to """!""",
+            // and the others
+            TokenCategorySpecific.LITERAL to """-\d(\d)*|\d(\d)*|true|false""",
+            TokenCategorySpecific.TYPE_IDENTIFIER to """\u\w*""",
+            TokenCategorySpecific.VARIABLE_IDENTIFIER to """(\l|_)\w*""",
+            TokenCategorySpecific.WHITESPACE to """(\n|\r|\t| )*""",
+            TokenCategorySpecific.COMMENT to """#\N*""",
+        )
+
+    private val specialCharacterMap =
```

Comment:
Doesn't it clash with Filip's implementation? I see that he already has special symbols for \l, \u. I truly don't know what's the best place to have this stuff, maybe in Filip's part, as you don't have any logic around it at all? I'm not sure tho. And make sure that you agree on this special characters, as I don't see \d and \w in Filip's code.

Reply:
My oversight, \w and \d will be added.